File: lib/screens/auth/signup.dart
import 'package:droute_driver_frontend/screens/driver/enable_location.dart';
import 'package:flutter/material.dart';
import 'package:droute_driver_frontend/styles/color/app_color.dart';
import 'package:droute_driver_frontend/models/auth_model.dart';
import 'package:droute_driver_frontend/screens/auth/signin.dart';

class SignUpPage extends StatefulWidget {
  const SignUpPage({super.key});

  @override
  State<SignUpPage> createState() => _SignUpPageState();
}

class _SignUpPageState extends State<SignUpPage> {
  final AuthModel _authModel = AuthModel();
  final TextEditingController _fullNameController = TextEditingController();
  final TextEditingController _emailController = TextEditingController();
  final TextEditingController _mobileController = TextEditingController();
  final TextEditingController _passwordController = TextEditingController();
  final TextEditingController _confirmPasswordController =
      TextEditingController();

  bool _isLoading = false;
  String? _errorMessage;
  Color _errorMessageColor = Colors.red;

  void _clearControllers() {
    _fullNameController.clear();
    _passwordController.clear();
    _emailController.clear();
    _mobileController.clear();
    _confirmPasswordController.clear();
  }

  // Request OTP action
  Future<void> _requestOtp() async {
    if (_fullNameController.text.isEmpty ||
        _emailController.text.isEmpty ||
        _passwordController.text.isEmpty ||
        _mobileController.text.isEmpty ||
        _confirmPasswordController.text.isEmpty) {
      _errorMessage = "Please Enter All given Fields!";
      _errorMessageColor = Colors.red;

      return;
    }
    ;

    if (_passwordController.text != _confirmPasswordController.text) {
      setState(() {
        _errorMessage = "Passwords do not match!";
        _errorMessageColor = Colors.red;
      });
      return;
    }

    setState(() {
      _isLoading = true;
      _errorMessage = null;
    });

    try {
      final otpResponse = await _authModel.otpRequest(_emailController.text);
      print("signup sent otp" + otpResponse);
      // Handle OTP response and navigate to OTP page
      if (otpResponse.isNotEmpty) {
        setState(() {
          _errorMessage = "OTP sent successfully!";
          _errorMessageColor = Colors.green;
          _clearControllers();
        });
        Navigator.pushNamed(
          context,
          '/otp',
          arguments: {
            'fullName': _fullNameController.text,
            'email': _emailController.text,
            'mobile': _mobileController.text,
            'password': _passwordController.text,
            'otpResponse': otpResponse,
          },
        );
      } else {
        setState(() {
          _errorMessage = "OTP request failed!";
          _errorMessageColor = Colors.red;
        });
      }
    } catch (e) {
      setState(() {
        _errorMessage = "An error occurred: $e";
        _errorMessageColor = Colors.red;
      });
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColor.primaryColor,
      body: SafeArea(
        child: SingleChildScrollView(
          physics: BouncingScrollPhysics(),
          child: Column(
            children: [
              SizedBox(height: MediaQuery.of(context).size.height * 0.05),
              Center(
                child: Image.asset(
                  'assets/images/droute_logo.png',
                  width: 200,
                  height: 170,
                ),
              ),
              SizedBox(height: 30),
              Container(
                width: double.infinity,
                decoration: BoxDecoration(
                  color: Colors.white,
                  borderRadius: BorderRadius.only(
                    topLeft: Radius.circular(20),
                    topRight: Radius.circular(20),
                  ),
                ),
                padding: const EdgeInsets.only(
                  top: 16,
                  left: 16,
                  right: 16,
                  bottom: 50,
                ),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      "Sign UP",
                      style: const TextStyle(
                        fontSize: 24,
                        fontWeight: FontWeight.bold,
                      ),
                    ),
                    SizedBox(height: 10),
                    CustomText(text: "Full Name:"),
                    TextFormField(
                      controller: _fullNameController,
                      decoration: InputDecoration(
                        hintText: "Enter Your Fullname",
                      ),
                    ),
                    const SizedBox(height: 16),
                    CustomText(text: "Email:"),
                    TextFormField(
                      controller: _emailController,
                      decoration: InputDecoration(hintText: "Enter Email Id"),
                    ),

                    SizedBox(height: 10),
                    CustomText(text: "Mobile Number:"),
                    TextFormField(
                      controller: _mobileController,
                      decoration: InputDecoration(
                        hintText: "Enter mobile Number",
                      ),
                    ),
                    SizedBox(height: 10),
                    CustomText(text: "Password:"),
                    TextFormField(
                      controller: _passwordController,
                      obscureText: true,
                      decoration: InputDecoration(hintText: "Enter Password"),
                    ),
                    SizedBox(height: 10),
                    CustomText(text: "Confirm Password"),
                    TextFormField(
                      controller: _confirmPasswordController,
                      obscureText: true,
                      decoration: InputDecoration(
                        hintText: "Enter Confirm Password",
                      ),
                    ),
                    SizedBox(height: 10),

                    if (_errorMessage != null)
                      Padding(
                        padding: const EdgeInsets.symmetric(vertical: 10),
                        child: Text(
                          _errorMessage!,
                          style: TextStyle(color: _errorMessageColor),
                        ),
                      ),
                    _isLoading
                        ? Container(
                          width: double.infinity,
                          padding: const EdgeInsets.symmetric(
                            vertical: 10,
                            horizontal: 20,
                          ),
                          decoration: BoxDecoration(
                            color: AppColor.primaryColor,
                            borderRadius: BorderRadius.circular(10),
                          ),
                          child: const Center(
                            child: CircularProgressIndicator(
                              color: Colors.white,
                            ),
                          ),
                        )
                        : InkWell(
                          onTap: _requestOtp,
                          child: Container(
                            width: double.infinity,
                            padding: const EdgeInsets.symmetric(
                              vertical: 10,
                              horizontal: 20,
                            ),
                            decoration: BoxDecoration(
                              color: AppColor.primaryColor,
                              borderRadius: BorderRadius.circular(10),
                            ),
                            child: const Text(
                              "Request OTP",
                              style: TextStyle(color: Colors.white),
                              textAlign: TextAlign.center,
                            ),
                          ),
                        ),

                    SizedBox(height: 16),
                    InkWell(
                      onTap: () {
                        Navigator.push(
                          context,
                          MaterialPageRoute(builder: (context) => SignInPage()),
                        );
                      },
                      child: Row(
                        mainAxisAlignment: MainAxisAlignment.center,
                        children: [
                          Text(
                            "Already Have an Account?",
                            style: TextStyle(fontWeight: FontWeight.bold),
                          ),
                          SizedBox(width: 5),
                          Text(
                            "Sign In",
                            style: TextStyle(
                              color: AppColor.primaryColor,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                        ],
                      ),
                    ),
                  ],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

Widget CustomText({required String text}) {
  return Text(
    text,
    style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16),
  );
}

File: src/screens/auth/signUp.js
// SignUpScreen.js
import React, { useState } from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  SafeAreaView,
  Image,
  KeyboardAvoidingView,
  Platform,
  ScrollView,
} from 'react-native';
import { Colors,commonStyles,Fonts } from '../../constants/styles';
import { authInput, authPassword } from '../../components/commonComponents';

const SignUpScreen = ({ navigation }) => {
 
  const [userType, setUserType] = useState('user');
  const [email, setEmail] = useState(null);
  const [fullName, setFullName] = useState(null);
  const [mobileNumber, setMobileNumber] = useState(null);
  const [password, setPassword] = useState(null);
  const [confirmPassword, setConfirmPassword] = useState(null);
  const [secureText, setSecureText] = useState(true);
  const [secureConfirmText, setSecureConfirmText] = useState(true);

  const handleSignUp = () => {
    // Validate form
    // if (!email || !mobileNumber || !password || !confirmPassword || !fullName) {
    //   alert('Please fill in all fields');
    //   return;
    // }
    
    // if (password !== confirmPassword) {
    //   alert('Passwords do not match');
    //   return;
    // }
  
    
    //  alert('Sign up successful!');
     navigation.navigate('VerificationScreen');
  };

  const navigateToSignIn = () => {
     navigation.navigate('SignInScreen');
   
  };

  return (
    <SafeAreaView style={styles.container}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={styles.keyboardAvoidingView}
      >
        <ScrollView contentContainerStyle={styles.scrollView}>
          <View style={styles.logoContainer}>
             <Image
              source={require("../../../assets/icon.png")}
              style={styles.logo}
              resizeMode="contain"
            />
          </View>
          <View style={styles.formContainer}>
            <Text style={styles.title}>Sign Up</Text>
        
            {authInput('Full Name', fullName, setFullName, 'Enter Full Name',"default")}
            {authInput('Email', email, setEmail, 'Enter Email Id Here', "email")}
            {authInput('Mobile Number', mobileNumber, setMobileNumber, 'Enter Mobile Number Here', "number")}
            {authPassword("Password",password, setPassword, 'Enter Password',secureText,setSecureText)}
            {authPassword("Confirm Password",confirmPassword, setConfirmPassword, 'Confirm your Password',secureConfirmText,setSecureConfirmText)}

            <TouchableOpacity style={styles.signUpButton} onPress={handleSignUp}>
              <Text style={styles.signUpButtonText}>Sign Up</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.signInLink} onPress={navigateToSignIn}>
              <Text style={styles.signInText}>
                Already Have an Account? <Text style={styles.signInHighlight}>Sign In</Text>
              </Text>
            </TouchableOpacity>
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
 container: {
    flex: 1,
  backgroundColor: Colors.primaryColor,
  },
  keyboardAvoidingView: {
    flex: 1,
  },
  scrollView: {
    flexGrow: 1,
  },
  logoContainer: {
    backgroundColor: Colors.primaryColor,
    paddingVertical: 40,
    alignItems: 'center',
    justifyContent: 'center',
     flex: 1, 
  },
  logo: {
    width: 200,
    height:200,
  },
  formContainer: {
    backgroundColor: Colors.whiteColor,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    marginTop: -20,
    paddingHorizontal: 24,
    paddingTop: 30,
    paddingBottom: 40,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    color: Colors.blackColor,
    marginBottom: 20,
  },
  signUpButton: {
   ...commonStyles.button,
      marginTop:10
  },
  signUpButtonText: {
   ...commonStyles.buttonText,
  },
  signInLink: {
    marginTop: 20,
    alignItems: 'center',
  },
  signInText: {
    fontSize: 14,
    color: Colors.blackColor,
  },
  signInHighlight: {
    color: Colors.primaryColor,
    fontWeight: 'bold',
  },
});

export default SignUpScreen;